internal/tiltfile/k8s.go

```diff
@@ -111,17 +111,27 @@ func (r k8sResource) refSelectorList() []string {
 
 func (s *tiltfileState) k8sYaml(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
 	var yamlValue starlark.Value
+
 	if err := s.unpackArgs(fn.Name(), args, kwargs,
 		"yaml", &yamlValue,
 	); err != nil {
 		return nil, err
 	}
 
-	entities, err := s.yamlEntitiesFromSkylarkValueOrList(thread, yamlValue)
-	if err != nil {
-		return nil, err
+	value := starlarkValueOrSequenceToSlice(yamlValue)
+	if len(value) > 0 {
+		entities, err := s.yamlEntitiesFromSkylarkValueOrList(thread, yamlValue)
+		if len(entities) == 0 {
```

Comment:
i think you just need to swap this `if` statement with the `if err != nil` below it.

A good rule of thumb in Go is to always handle the error first, then handle the other cases!

Reply:
I will ink this in mind because I did just this and all the tests passed locally woohooo!